File: python/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            compared = target - nums[i]
            for j in range(i + 1, len(nums)):
                if nums[j] == compared:
                    return [i, j]
        return []


#test

result = Solution()
print(result.twoSum([2,4,5,6,5,2], 11))
                

File: rust/two_sum.rs
use std::collections::HashMap;

pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut map = HashMap::new();

    for(i, &num) in nums.iter().enumerate() {
        let complement = target - num;

        if let Some(&j) = map.get(&complement) {
            return vec![j as i32, i as i32];
        }
        map.insert(num, i);
    }
    vec![]
}

fn main() {
    let test: Vec<i32> = vec![2,3,5,7,9];
    let target: i32 = 8;
    println!("{:?}", two_sum(test, target))
}
